torch/_refs/__init__.py

```diff
@@ -4313,8 +4316,21 @@ def lerp(start: Tensor, end: Tensor, weight: Union[Tensor, NumberType]):
     mask = weight.abs() >= 0.5
     coeff = torch.where(mask, weight - 1, weight)
     base = torch.where(mask, end, start)
-    return coeff * (end - start) + base
-
+    output = coeff * (end - start) + base
+    # make sure the decomposition output's stride is same as non-decomposition path.
+    strides = utils.compute_elementwise_output_strides(*inputs)
+    if output.stride() != strides:
+        new_output = torch.empty_strided(
+            output.shape,
+            strides,
+            dtype=output.dtype,
+            layout=output.layout,
+            device=output.device,
+            requires_grad=output.requires_grad,
+        )
+        new_output.copy_(output)
```

Comment:
we cannot use inplace ops in decompositions of non-inplace ops, as this will throw off functionalization. @bdhirsh what's the correct way of doing this?

Reply:
Probably do `output.as_strided(output.shape, strides)`

Reply:
I think we need a new `prim.copy_strided` which is the functional equivalent. That said, is `compute_elementwise_output_strides` safe to use inside a decomposition since the output depends on the stride values?

Reply:
Two other options I can think of:

(1) change this decomp to an inductor lowering

(2) we have an existing `aten.copy()`. We could... change its semantics to preserve input strides. That shouldn't be too BC breaking, since nobody should really be using that op outside of post-aot autograd backends (inductor)

Reply:
> change this decomp to an inductor lowering

Note that this decomposition isn't being used by inductor, but it is used as the meta-implementation which is problematic for tracing.

Reply:
How about changing this?
```
    strides = utils.compute_elementwise_output_strides(*_maybe_broadcast(*inputs))
    if output.stride() != strides:
        new_output = torch.empty_strided(
            output.shape,
            strides,
            dtype=output.dtype,
            layout=output.layout,
            device=output.device,
            requires_grad=output.requires_grad,
        )

        new_output = utils.clone_preserve_strides(new_output)
        new_output.copy_(output)
        output = new_output
    return output
```

As copy implementation in https://github.com/pytorch/pytorch/blob/c47bdd7522c7146c091c95ff5b4ce354d7a8870c/aten/src/ATen/native/Copy.cpp#L282-L288.

Reply:
@peterbell10, there has a **prim.copy_to**: https://github.com/pytorch/pytorch/blob/10910758f462a7b673b60c4755c553ba2abb462d/torch/_prims/__init__.py#L2176, could we directly use it()? if ok, we need to register the lowering path at inductor side.

Reply:
> there has a prims.copy_to

`prims.copy_to` also does inplace copy, not functional...

Reply:
btw, is this a general problem (not specific to `lerp`) that the output of a decomposition needs to be the same as the non-decomposed one? If so, do we need something like a decorator to specify the expected output strides (e.g., the output strides of lerp should follow `start`), not add a post-processing op-by-op?

Reply:
> For prims.copy_strided, do you mean to add a copy variant with an additional strides as input?

Yes, exactly. `prims.copy_strided(x, stride=output_strides)` which returns a new tensor with the specified strides and all other properties copied from `x`.

> btw, is this a general problem

Yes, `torch/_meta_registrations.py` automatically registers decompositions as the meta implementation, which means the strides really do matter now. 

> do we need something like a decorator to specify the expected output strides (e.g., the output strides of lerp should follow start), not add a post-processing op-by-op?

I think a wrapper that automatically computes the strides using `compute_elementwise_output_strides` would be really useful.